### PR TITLE
Add LORE estimator: low-rank ordinal embedding with intrinsic dimensionality estimation

### PR DESCRIPTION
**Pull request by vivek2000anand**
## Summary

  This PR adds LORE (Low-Rank Ordinal Embedding) as a new CBlearn estimator. LORE jointly learns an ordinal embedding and estimates its intrinsic dimensionality by applying
  Schatten-p (or nuclear norm) regularisation to the singular values of the embedding matrix. The estimated intrinsic rank is exposed via the `rank_` attribute.

  Reference: Anand, V., Helbling, A., Davenport, M. A., Berman, G. J., Alagapan, S., & Rozell, C. J. (2026). LORE: Jointly Learning The Intrinsic Dimensionality and Relative
   Similarity Structure from Ordinal Data. ICLR 2026. https://arxiv.org/abs/2602.04192
```
@inproceedings{anand2026lore,
  title={LORE: Jointly Learning The Intrinsic Dimensionality and Relative Similarity Structure from Ordinal Data},
  author={Anand, Vivek and Helbling, Alec and Davenport, Mark A. and Berman, Gordon J and Alagapan, Sankaraleengam and Rozell, Christopher J},
  paper_url = {https://arxiv.org/abs/2602.04192},
  booktitle={International Conference on Learning Representations (ICLR)},
  month        = {April},
  address      = {Rio de Janeiro, Brazil},
  year={2026}
}
```

## Changes

  - **`_lore.py`**: `LORE(BaseEstimator, TripletEmbeddingMixin)` — follows the same pattern as SOE, CKL, FORTE. Supports `backend="torch"` (default, GPU-capable) and
  `backend="scipy"` (pure NumPy/SciPy, no GPU required). Exposes `rank_`, `stress_`, `n_iter_`, `sigma_history_`, `objectives_` after fitting.
  - **`_lore_utils.py`**: Self-contained training backends. No dependency on any external package beyond NumPy/SciPy (scipy backend) or PyTorch (torch backend). The
  Lipschitz constant is estimated via the power method in both backends.
  - **`test_lore.py`**: 19 tests covering both backends, rank reduction on low-dimensional data, analytic gradient check (`scipy.optimize.check_grad`), monotone convergence,
   all fitted attributes, and an end-to-end demo workflow. 98% coverage on `_lore.py`, 92% on `_lore_utils.py`.

## Key design decisions

  - `n_components` is an *over-estimate* of the ambient dimension; LORE discovers the true rank via regularisation
  - `lamb=0.01` matches the paper's default; `p=0.5` is the Schatten-p exponent
  - The Lipschitz constant `mu` is estimated via power iteration in both backends (not a heuristic)
  - All torch imports are lazy (inside functions) so the scipy backend works without PyTorch

## Test results

  pytest cblearn/embedding/tests/test_lore.py -v   # 19/19 passed
  pytest cblearn/embedding/tests/test_sklearn_estimator_checks.py -v  # 267/267 passed
  pytest cblearn/ -q  # 136 passed, 0 failed

copilotted with Claude Code

### CODE REVIEW

**Review by vivek2000anand — COMMENTED** [review]
Added a new Ordinal Embedding, LORE, to cblearn. Nature Vogue Similarity no longer has a download link working so took it out. Material Similarity checksum has changed. Fixed issues with R conversion and Sklearn compatibility.

**Review by guillermoaguilar — APPROVED** [review]

**Comment by guillermoaguilar** [comment]
@vivek2000anand :  I have now write access to the repository. I have tested your fork in my machine and 1 test was hanging because the dataset it was trying to fetch does not exist anymore. I have committed over your branch. I will also test for newer python version once I'm on it. My goal is to merge to main only when all tests are passing.

**Comment by guillermoaguilar** [comment]
Ok so I have done all necessary changes to bring cblearn up to date (in terms of dependencies) in another PR (#84), which I have just merged. All tests are passing now there. 

I did this in order to separate what is mainly "maintenance" of the code versus the addition of new features, such as here with LORE.

Now the next step is to fix this branch so that only the new components (LORE) are present. I will force-push these changes to this branch.

**Comment by guillermoaguilar** [comment]
This PR has been divided into two: #84  brings cblearn up to date, and #88  adds LORE. This PR is now obsolete.
